apps/analog-app/src/app/pages/newsletter.server.ts

```diff
@@ -0,0 +1,53 @@
+import { PageServerAction } from '@analogjs/router';
+import { readFormData } from 'h3';
+
+import type { ActionResult } from './form-action.directive';
+
+function fail(status: number, errors: object) {
+  return new Response(JSON.stringify(errors), {
+    status,
+    headers: {
+      'X-Analog-Errors': 'true',
+    },
+  });
+}
+
+function json(data: object, status = 301) {
+  return new Response(JSON.stringify(data), {
+    status,
+  });
+}
+
+function redirect(url: string) {
+  return new Response(null, {
+    status: 301,
+    headers: {
+      'X-Analog-Redirect': url,
+    },
+  });
+}
+
+export function load() {
+  return {
+    loaded: true,
+  };
+}
+
+export async function action({
+  event,
+}: PageServerAction): Promise<ActionResult> {
+  const body = await readFormData(event);
+  const email = body.get('email') as string;
+
+  if (!email) {
+    return fail(422, { errors: { email: 'Email is required' } });
```

Comment:
question: If JavaScript is disabled, what happens in this case? Can we set up something like `[...not-found].page.ts` but like `[...422].page.ts` to handle this?

Reply:
I tried handling it manually by adding a `format: 'json'` field if the call is made via Js:
- https://github.com/ajitzero/spartan-repro/blob/feat/form-action/analog-app/src/server/routes/v1/subscribe.ts
- https://github.com/ajitzero/spartan-repro/blob/feat/form-action/analog-app/src/app/pages/subscribe/(home).page.ts#L54

Reply:
This is still TBD. Ideally, if JS is off, you come back to this page with errors displayed as part of the rendered HTML